packages/number-input/src/Input/Input.tsx

```diff
@@ -0,0 +1,184 @@
+  inputThemeStyles,
+  selectBaseStyles,
+  sizeInputStyles,
+  warningIconClassName,
+  wrapperBaseStyles,
+  wrapperClassName,
+  wrapperDisabledStyles,
+  wrapperErrorStyles,
+  wrapperHoverStyles,
+  wrapperStateStyles,
+  wrapperThemeStyles,
+} from './Input.styles';
+import { InputProps } from './Input.types';
+
+/**
+ * @internal
+ */
+
+export function Input({
+  value: valueProp,
+  onChange: onChangeProp,
+  disabled,
+  size = Size.Default,
+  hasSelectOptions,
+  state = State.None,
+  errorMessage,
+  ...rest
+}: InputProps) {
+  const inputRef = useRef<HTMLInputElement | null>(null);
+  const { theme } = useDarkMode();
+
+  const renderErrorIcon = state === State.Error;
+
+  const { value, handleChange: handleChangeProp } = useControlledValue(
+    valueProp,
+    onChangeProp,
+  );
+
+  const handleChange = (e: React.ChangeEvent<HTMLInputElement>) => {
+    handleChangeProp?.(e);
+  };
+
+  /**
+   * Custom arrow buttons do not trigger a change event on the input.
+   * Creating a synthetic event will trigger a new change event.
+   */
+  const handleSyntheticEvent = () => {
+    const nativeEvent = new Event('change', { bubbles: true });
+    const synthEvent = createSyntheticEvent(nativeEvent, inputRef.current!);
+    handleChange?.(synthEvent as ChangeEvent<HTMLInputElement>);
+  };
+
+  const handleIncrementClick = () => {
+    if (!disabled) {
+      inputRef.current?.stepUp();
+      handleSyntheticEvent();
+    }
+  };
+
+  const handleDecrementClick = () => {
+    if (!disabled) {
+      inputRef.current?.stepDown();
+      handleSyntheticEvent();
+    }
+  };
+
+  /**
+   * Edge case if the user clicks on an arrow button then switches to keyboard click.
+   * By default if focus is in the input then the keyboard clicks will work automatically but since the buttons are custom and outside of the input we are mimicking native behavior.
+   */
+  const handleArrowKeyDown = (e: React.KeyboardEvent) => {
+    if (!disabled) {
+      if (e.key === 'ArrowUp') {
+        e.preventDefault();
+        handleIncrementClick();
+      }
+
+      if (e.key === 'ArrowDown') {
+        e.preventDefault();
+        handleDecrementClick();
+      }
+    }
+  };
+
+  return (
+    <div
+      aria-disabled={disabled}
+      className={cx(
+        wrapperClassName,
+        wrapperBaseStyles,
+        wrapperThemeStyles[theme],
+        wrapperStateStyles[theme][state],
+        {
+          [wrapperHoverStyles[theme][state]]: !disabled,
+          [wrapperDisabledStyles[theme]]: disabled,
+          [selectBaseStyles]: hasSelectOptions,
+          [wrapperErrorStyles]: renderErrorIcon,
+        },
+      )}
+    >
+      <input
+        ref={inputRef}
+        className={cx(
+          inputBaseStyles,
+          inputThemeStyles[theme],
+          sizeInputStyles[size],
+          {
+            [errorInputStyles[size]]: renderErrorIcon,
+            [inputAnimateStyles]: !disabled,
+          },
+        )}
+        type="number"
```

Comment:
should we add a console warning to our text-input if a user tries to use `type="number"` there?